build/changeSets/2019/convertUitype20to19.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/*************************************************************************************************
+ * Copyright 2018 JPL TSolucio, S.L. -- This file is a part of TSOLUCIO coreBOS Customizations.
+* Licensed under the vtiger CRM Public License Version 1.1 (the "License"); you may not use this
+* file except in compliance with the License. You can redistribute it and/or modify it
+* under the terms of the License. JPL TSolucio, S.L. reserves all rights not expressly
+* granted by the License. coreBOS distributed by JPL TSolucio S.L. is distributed in
+* the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. Unless required by
+* applicable law or agreed to in writing, software distributed under the License is
+* distributed on an "AS IS" BASIS, WITHOUT ANY WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific language governing
+* permissions and limitations under the License. You may obtain a copy of the License
+* at <http://corebos.org/documentation/doku.php?id=en:devel:vpl11>
+*************************************************************************************************/
+
+class convertUitype20to19 extends cbupdaterWorker {
+
+	public function applyChange() {
+		global $adb;
+		if ($this->hasError()) {
+			$this->sendError();
+		}
+		if ($this->isApplied()) {
+			$this->sendMsg('Changeset '.get_class($this).' already applied!');
+		} else {
+            $tabid = getTabid('Faq');
+            $this->ExecuteQuery("UPDATE vtiger_field SET uitype = 19 WHERE tablename = 'vtiger_faq' and tabid =? and (columnname = ? or columnname = ?)", array($tabid, 'question', 'answer'));
```

Comment:
don't we have to eliminate the uitype from webservice?
wouldn't it be better to not be so specific? I mean, imagine a coreBOS install where someone has created a new field of uitype 20, with your change that field would not be updated and would stop working. I think it would be better to just change all the 20 to 19 regardless of what module they are on.  Do you agree?

Reply:
yes I agree that mean this was wrong because it only look at module FAQ mean specific to only module